include/hip/math_constants.h

```diff
@@ -0,0 +1,39 @@
+/*
+Copyright (c) 2015 - present Advanced Micro Devices, Inc. All rights reserved.
+
+Permission is hereby granted, free of charge, to any person obtaining a copy
+of this software and associated documentation files (the "Software"), to deal
+in the Software without restriction, including without limitation the rights
+to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+copies of the Software, and to permit persons to whom the Software is
+furnished to do so, subject to the following conditions:
+
+The above copyright notice and this permission notice shall be included in
+all copies or substantial portions of the Software.
+
+THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+THE SOFTWARE.
+*/
+
+#ifndef HIP_INCLUDE_HIP_MATH_CONSTANTS_H
+#define HIP_INCLUDE_HIP_MATH_CONSTANTS_H
+
+// Some standard header files, these are included by hc.hpp and so want to make them avail on both
+// paths to provide a consistent include env and avoid "missing symbol" errors that only appears
+// on NVCC path:
+
+
+#if defined(__HIP_PLATFORM_HCC__) && !defined(__HIP_PLATFORM_NVCC__)
+#include <hip/hcc_detail/math_constants.h>
+#elif defined(__HIP_PLATFORM_NVCC__) && !defined(__HIP_PLATFORM_HCC__)
+//#include <hip/nvcc_detail/math_constants.h>
```

Comment:
There needs to be a nvcc_detail/math_constants.h that maps the HIPRT_* constants to the CUDA equivalent ones.

Reply:
@mangupta 

The CUDA equivalent of "math_constants.h" is part of the CUDA header files (i.e. it is already present at  `<cuda_install_dir>/include/math_constants.h`)

Do we still need to have a `nvcc_detail/math_constants.h`? If yes, then can we simply copy the one from the CUDA install?

Reply:
@deven-amd Maybe I wasn't clear enough.
Say there is a HIP sample that uses ```HIPRT_INF_F```. When compiling on HIP_PLATFORM_HCC, the macro gets resolved via ```hip/hcc_detail/math_constants.h```. However the same macro will not get resolved when compiling on HIP_PLATFORM_NVCC. What needs to happen is that there needs to be a ```hip/nvcc_detail/math_constants.h``` that maps the HIPRT_* macros to equivalent CUDART_* so that compilation works on both platforms.

Reply:
Ok, now I see what you mean...let me make the update and resubmit this PR.

Reply:
BTW, what is workflow to submit a second PR from my fork, while another is still outstanding (#541)